### DISCUSSION TITLE
Add Event Listener for invalidating queries

### DISCUSSION THREAD

**Discussion by paulwer** [TanStack/query#7955] [Ideas] [answered]
Hi everyone, we are using this library in combination of dataloader to combine multiple requests into one, which also leverages a small caching solution. When invalidating a query-key within vue-query we want to automatic invalidate the cache on the external lib also.

Therefore I think it would be great to define an event listener for such use-cases.

Thanks
Paul

**Comment by TkDodo** [comment]
you should get an event of `type: 'fetch'` when the query refetches (I'm talking about the [cache level subscriptions](https://tanstack.com/query/v5/docs/reference/QueryCache/#querycachesubscribe))

**Comment by paulwer** [comment, marked as answer]
gotcha, thank you, that worked for us:
![image](https://github.com/user-attachments/assets/3f7d2c5e-009d-4771-ba17-1e52a565249a)
